packages/ts-transform/src/utils/css-transform.tsx

```diff
@@ -0,0 +1,89 @@
+import postcss, { plugin } from 'postcss';
+import autoprefixer from 'autoprefixer';
+import cssnano from 'cssnano-preset-default';
+import nested from 'postcss-nested';
+import whitespace from 'postcss-normalize-whitespace';
+
+const minify = () => {
+  const preset = cssnano();
+  // We exclude async because we need this to run synchronously as ts transformers aren't async!
+  const asyncPluginsToExclude = ['postcss-svgo'];
+
+  return preset.plugins
+    .map(([creator]: any) => {
+      // replicate the `initializePlugin` behavior from https://github.com/cssnano/cssnano/blob/a566cc5/packages/cssnano/src/index.js#L8
+      return creator();
+    })
+    .filter((plugin: any) => {
+      return !asyncPluginsToExclude.includes(plugin.postcssPlugin);
+    });
+};
+
+const parentOrphenedPseudos = plugin('parent-orphened-pseudos', () => {
+  return root => {
+    root.walkRules(rule => {
+      if (rule.selector.includes(':')) {
+        const newSelector = rule.selector
+          .split(',')
+          .map(part => {
+            if (part.match(/^. /)) {
+              // If the selector has one characters with a space after it, e.g. "> :first-child" then return early.
+              return part;
+            }
+
+            if (part.includes(':')) {
+              // If the selector starts with a colon or has a space and then a colon prepend an "&"!
+              return part.replace(/^:| :/g, '&:');
```

Comment:
do you know for what input you can receive a colon preceded with whitespace?

Reply:
it was for one of the tests i saw - i'd have to remove this and see which test fails 😅

Reply:
could you check this out? would love to see when this happens, probably it's related to how postcss work and what output does it yield, but knowing the test case I could recheck if this is working as intended

Reply:
turns out i just had to parse away the unneeded whitespace and then its not a problem.

Reply:
Yes, rule elements have their computed selectors available in `.props` property. This holds already computed (expanded) selectors, so for this input:
```sass
a,b,c{d,e{color:hotpink;}}
```
it would have such props:
```js
['a d','b d','c d','a e','b e','c e']
```
if you want to get "selectors" only for the "current level" then you need to reparse `.value` which for the inner node here would hold `'d,e'` string.

Reply:
man that would be sweet - I don't think the current version of Post CSS does that....